Sources/TuistGenerator/Mappers/SynthesizedResourceInterfaceProjectMapper.swift

```diff
@@ -193,25 +174,32 @@ public final class SynthesizedResourceInterfaceProjectMapper: ProjectMapping { /
         path: AbsolutePath,
         target: Target
     ) -> String {
-        switch resourceSynthesizer.parser {
-        case .assets, .strings, .fonts, .interfaceBuilder:
-            return target.name.camelized.uppercasingFirst
-        case .plists, .json, .yaml, .coreData:
-            return path.basenameWithoutExt.camelized.uppercasingFirst
-        }
+        return target.name.camelized.uppercasingFirst
     }
 
     private func paths(
         for resourceSynthesizer: ResourceSynthesizer,
-        target: Target
+        target: Target,
+        developmentRegion: String?
     ) -> [AbsolutePath] {
         let resourcesPaths = target.resources
             .map(\.path)
 
         var seen: Set<String> = []
-        return resourcesPaths
+        var paths = resourcesPaths
             .filter { $0.extension.map(resourceSynthesizer.extensions.contains) ?? false }
             .filter { seen.insert($0.basename).inserted }
+            .sorted()
+
+        if resourceSynthesizer.isLocalizable {
+            let developmentRegion = developmentRegion ?? "en"
+            // Let's sort paths moving the development region localization's one at first
+            paths.sort(by: { lhs, rhs in
+                lhs.pathString.contains(developmentRegion) && !rhs.pathString.contains(developmentRegion)
```

Comment:
this could lead to changes that will be dependent on the user's environment -> e.g. I can have the project in `/Users/en/...` and someone else in `/Users/cz/...` where user "en" would always contain the `developmentRegion`. I'd therefore sort it by `basename` instead of `pathString`

Reply:
I'd also maybe prefer `paths.filter { $0.basename.contains(developmentRegion) } + paths.filter { !$0.basename.contains(developmentRegion) }`. To me it's a little bit more clearer - but the way you wrote it is fine as well, especially with the comment 👍

Reply:
[here](https://github.com/tuist/tuist/blob/main/Sources/TuistGenerator/Generator/ProjectDescriptorGenerator.swift#L153) we also factor in that the base lproj files could be `Base.lproj` instead of `$DEVELOPMENT_LANGUAGE.proj`.

Reply:
Added a queue to sort files based on: [Optional(developmentRegion), Base, en] so that base if exists takes priority over en and development region above all of them if it was specified

Reply:
Unfortunately though basename does not contain the `.lproj` part so it can't be used like this, what I did was to access the `parentDirectory().basename` to perform such checks, do you think it will be enough? Is there any scenario for which the .lproj dir can be even at a higher level?